Package: All Versions / Recent Version buttons

eksperimental opened the pull request:
It changes the previous behaviour to the text is toggled on click.

Closes: https://github.com/hexpm/hex_web/issues/143

ericmj commented on lib/hex_web/web/templates/main.html.eex:
```diff
@@ -73,6 +73,7 @@
 
     <script src="//cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
     <script src="//cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.1/js/bootstrap.min.js"></script>
+    <script src="/static/main.js?<%= asset_id %>"></script>
```
What is this for?

eksperimental replied:
sorry that file was not staged,
I have updated the commit, let me know if you want the code inside the template.

ericmj commented:
Can we keep it uncollapsed for people without javascript?

eksperimental commented:
sure! but then I think I need to refactor the whole thing, as it will be better to have one list, and just toggle whatever elements we don't want to show if javascript is available.

ericmj commented:
Ping? :)

eksperimental commented:
sorry about the delay @ericmj , I will get my hand on this tomorrow (I am finishing something for Elixir 1.1.0)

eksperimental commented:
done :-)

henrik commented on priv/static/static/main.js:
```diff
@@ -0,0 +1,11 @@
+$('.show-versions .invisible').removeClass('invisible').toggle();
+$('.show-versions .toggle-text').click(function() {
+  $(this).find('span').each(function() {
+    $(this).toggle();
```
I think `$(this).find('span').toggle();` would also work here.

eksperimental replied:
thanks @henrik, I have updated the code.

ericmj commented:
Thank you @eksperimental. Will test this locally and then merge.

eksperimental commented:
I have updated the code and rebased it with the current master,
including suggestion by @henrik and also by default the show all versions button is disabled, only shown when JS is enabled.

@ericmj the build is failing with elixir master on OTP 17.x,
locally runnning elixir 1.1.0-beta on OTP 18 pass all tests including integration

Could be a non-backward compatible changed recently introduced in elixir master when running OTP 17?

ericmj commented:
Thank you. Will test this locally.

ericmj commented:
Looked great. Thanks!
